Code/CryGame/HUD/HUDInterfaceEffects.cpp

```diff
@@ -1593,20 +1622,34 @@ void CHUD::ShowProgress(int progress, bool init /* = false */, int posX /* = 0 *
 
 		m_iProgressBarX = posX;
 		m_iProgressBarY = posY;
-		m_sProgressBarText = string(text);
+		if (text.data())
+		{
+			m_sProgressBarText = (string)text.data();
+		}
```

Comment:
This is problematic because it doesn't clear `m_sProgressBarText` when `text` is completely empty. Also, `.empty()` should be used in conditions.

If `m_sProgressBarText` is `std::string`, then you can do just:
```suggestion
		m_sProgressBarText = text;
```